tests/src/IrcMessageHandler.cpp

```diff
@@ -603,7 +603,11 @@ TEST_P(TestIrcMessageHandlerP, Run)
 
     delete ircMessage;
 
-    ASSERT_TRUE(snapshot->run(got, UPDATE_SNAPSHOTS));
+    ASSERT_TRUE(snapshot->run(got, UPDATE_SNAPSHOTS))
+        << "Snapshot " << snapshot->name() << " failed. Expected JSON to be\n"
+        << QJsonDocument::fromVariant(snapshot->output().toVariant()).toJson()
```

Comment:
`fromVariant` can only create objects or arrays.
This should be something like

```cpp
QString debugRepr(const auto& it) {
    QString str;
    QDebug dbg(&str);
    dbg << it;
    return str;
}
```

Qt 6.9 will allow `QJsonValue.toJson()`.